src/share/cxx/Tracers.cpp

```diff
@@ -21,6 +21,7 @@ Tracers::Tracers(const int num_elems, const int num_tracers)
   qdp = decltype(qdp)("tracers mass", num_elems);
   qtens_biharmonic = decltype(qtens_biharmonic)("qtens(_biharmonic)", num_elems);
   qlim = decltype(qlim)("qlim", num_elems);
+  fq = decltype(fq)("fq", num_elems);
```

Comment:
We should avoid alloc'ing such a large array unless it's needed. For ftype==2, the default in E3SM, it is not needed. So here I would put in an ftype check.

Reply:
I'll see what I can do

Reply:
Does it make sense to put ftype in SimulationParams.hpp? We'll need it in several spots, I believe.

Reply:
We can put ftype inside SimulationParams...possibly as an enum, rather than an int...